### PR TITLE
Check if the newly set password is the same as the old one

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

This is especially worth adding in the context of forced passwords (via env var) as the password is always "set" on each config change. This PR effectively prevents FTL from wiping all sessions and forcing the user to re-login when changing even settings which are 100% unrelated to the API.

---

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

File: src/config/password.c [hunk 1]
```diff
@@ -601,6 +601,13 @@ int run_performance_test(void)
 
 bool set_and_check_password(struct conf_item *conf_item, const char *password)
 {
+	// Check if the newly set password is the same as the old one
+	if(verify_password(password, config.webserver.api.pwhash.v.s, false) == PASSWORD_CORRECT)
```
Comment (yubiuser) [inline src/config/password.c:605]: Do we also need to check if no password is set (or empty) to exit the check early?

**Comment by rdwebdesign** [comment]
Do we still need this check?

https://github.com/pi-hole/FTL/blob/2406e1a70ee9ef83b98ed672df2705d02b016da2/src/api/config.c#L289-L294

File: src/config/password.c (hunk 1 above)
Comment (rdwebdesign) [reply src/config/password.c:605]: This is already checked here:

https://github.com/pi-hole/FTL/blob/2406e1a70ee9ef83b98ed672df2705d02b016da2/src/config/password.c#L611-L621

Comment (DL6ER) [reply src/config/password.c]: I agree it's worthwhile to do this before investing any work in trying to validate a hash (it takes a lot of time). I have added and pushed this.

**Review by yubiuser — APPROVED** [review]

**Pull request merged**